web/blueprints/facilities/tables.py

```diff
@@ -1,77 +1,58 @@
 
-from web.blueprints.helpers.table import BootstrapTable, Column
+from web.blueprints.helpers.table import BootstrapTable, Column, LinkColumn, \
+    button_toolbar
 
 
 class SiteTable(BootstrapTable):
-    def __init__(self, *a, **kw):
-        super().__init__(*a, columns=[
-            Column('site', 'Site', formatter='table.linkFormatter'),
-            Column('buildings', 'Buildings', formatter='table.multiBtnFormatter'),
-        ], **kw)
+    site = LinkColumn("Site")
+    buildings = Column("Buildings", formatter='table.multiBtnFormatter')
 
 
 class BuildingLevelRoomTable(BootstrapTable):
-    def __init__(self, *a, **kw):
-        super().__init__(*a, columns=[
-            Column('room', 'Raum', formatter='table.linkFormatter'),
-            Column('inhabitants', 'Bewohner', formatter='table.multiBtnFormatter'),
-        ], table_args={
+    class Meta:
+        table_args = {
             'data-sort-name': 'room',
             'data-query-params': 'perhaps_all_users_query_params',
-        }, **kw)
+        }
+
+    room = LinkColumn("Raum")
+    inhabitants = Column('Bewohner', formatter='table.multiBtnFormatter')
 
-    def generate_toolbar(self):
-        """Generate a toolbar with a "Display all users" button
-        """
-        yield '<a href="#" id="rooms-toggle-all-users" class="btn btn-default" role="button">'
-        yield '<span class="glyphicon glyphicon-user"></span>'
-        yield 'Display all users'
-        yield '</a>'
+    toolbar = button_toolbar("Display all users", href="#",
+                             icon="glyphicon-user")
 
 
 class RoomLogTable(BootstrapTable):
-    def __init__(self, *a, **kw):
-        super().__init__(*a, columns=[
-            Column('created_at', 'Erstellt um', formatter='table.dateFormatter'),
-            Column('user', 'Nutzer', formatter='table.linkFormatter'),
-            Column('message', 'Nachricht'),
-        ], **kw)
+    created_at = Column("Erstellt um", formatter='table.dateFormatter')
+    user = LinkColumn("Nutzer")
+    message = Column("Nachricht")
 
 
 class RoomOvercrowdedTable(BootstrapTable):
-    def __init__(self, *a, **kw):
-        super().__init__(*a, columns=[
-            Column('room', 'Raum', formatter='table.linkFormatter'),
-            Column('inhabitants', 'Bewohner', formatter='table.multiBtnFormatter'),
-        ], table_args={
-            'data-sort-name': 'room',
-        }, **kw)
+    room = LinkColumn("Raum")
+    inhabitants = Column("Bewohner", formatter='table.multiBtnFormatter')
+
+    class Meta:
+        table_args = {'data-sort-name': 'room'}
 
 
 class PatchPortTable(BootstrapTable):
+    name = Column('Name', width=2)
+    room = Column('→ Raum', formatter='table.linkFormatter', width=5)
+    switch_port = Column('→ Switch-Port', formatter='table.linkFormatter', width=3)
+    edit_link = Column('Editieren', formatter='table.btnFormatter')
+    delete_link = Column('Löschen', formatter='table.btnFormatter')
```

Comment:
There should be used a `BtnColumn` here too.

Reply:
Resolved in 6b41f59866d27292d84120ba8391cb042618f21f